src/server/middleware/auth/authorizerbase.js

```diff
@@ -0,0 +1,122 @@
+/*globals*/
+/*jshint node:true*/
+/**
+ * @author pmeijer / https://github.com/pmeijer
+ */
+'use strict';
+
+var Q = require('q'),
+    ENTITY_TYPES = {
+        PROJECT: 'PROJECT',
+        USER: 'USER'
+    };
+
+/**
+ * @typedef {object} AccessRights
+ * @prop {boolean} read
+ * @prop {boolean} write
+ * @prop {boolean} delete
+ *
+ * @example
+ * {
+ *   read: true,
+ *   write: true,
+ *   delete: false
+ * }
+ */
+
+
+/**
+ *
+ * @param {GmeLogger} mainLogger
+ * @param {GmeConfig} gmeConfig
+ * @constructor
+ */
+function AuthorizerBase(mainLogger, gmeConfig) {
+
+    /**
+     * @type {{PROJECT: string, USER: string}}
+     */
+    this.ENTITY_TYPES = ENTITY_TYPES;
+
+    /**
+     * @type {GmeConfig}
+     */
+    this.logger = mainLogger.fork('AuthorizerBase');
+
+    /**
+     * @type {GmeLogger}
+     */
+    this.gmeConfig = gmeConfig;
+}
+
+/**
+ * @type {{PROJECT: string, USER: string}}
+ */
+AuthorizerBase.ENTITY_TYPES = ENTITY_TYPES;
+
+/**
+ *
+ * @param {object} params
+ * @param {object} params.collection - Mongo collection to default database.
```

Comment:
I agree that it's not the nicest solution, but since we have the connection initiated from gmeAuth (and will use it for the project metadata and to store user settings regardless of the chosen authorizer) it might as well be passed into the authorizer. It's up to any implementation to use it or not. The reason I made start/stop asynchronous is to give allow for custom authorizers to initiate connections or what ever they need to do.